### PR TITLE
Make Certificate Selectable in TestApp

### PR DESCRIPTION
**Pull request by wba2hi**
Closes: #15

### CODE REVIEW

File: app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/DataBrokerView.kt [hunk 1]
```diff
@@ -583,9 +420,10 @@ fun DataBrokerOutput(viewModel: OutputViewModel, modifier: Modifier = Modifier)
 @Composable
 fun Preview() {
     KuksaAppAndroidTheme {
+        val connectionInfoRepository = ConnectionInfoRepository(LocalContext.current)
         DataBrokerView(
             TopAppBarViewModel(),
-            ConnectionViewModel(LocalContext.current.applicationContext as Application),
+            ConnectionViewModel(connectionInfoRepository),
```
Comment (wba2hi) [inline app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/DataBrokerView.kt:426]: Required changes to get the Preview working again

File: kuksa-sdk/consumer-rules.pro [hunk 2]
```diff
@@ -0,0 +1,2 @@
+-keep class org.eclipse.kuksa.proto.v1.** { *; }
+-keep class com.google.protobuf.** { *; }
```
Comment (wba2hi) [inline kuksa-sdk/consumer-rules.pro:2]: redundant with your branch, but I had to add these rules again, to make sure running the app minified is working in the current state

File: app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileWriter
+
+class ConnectionInfoTest : BehaviorSpec({
+    given("A custom ConnectionInfo with a custom Certificate") {
+        // Uri is an android specific class, which does not exist in plain JUnit. Methods like Uri.parse will simply
+        // return null due to the addition of testOptions.unitTests.isReturnDefaultValues true
```
Comment (wba2hi) [inline app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt:32]: can only be fixed with Robolectric, but if I remember correctly there was an incompatibility of Kotest <-> Robolectric

Comment (Chrylo) [reply app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt:32]: Exactly

File: app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+package org.eclipse.kuksa.testapp.model
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.File
+import java.io.FileWriter
+
+class ConnectionInfoTest : BehaviorSpec({
```
Comment (Chrylo) [inline app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt:29]: We are missing a run config for the app unit tests, right?

File: app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                } catch (_: Exception) {
+                    isExceptionThrown = true
+                }
+            }
+
+            then("No Exception is thrown") {
+                isExceptionThrown shouldBe false
+            }
+
+            then("We are falling back to the defaultValue") {
```
Comment (Chrylo) [inline app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt:93]: We? You mean "it" ? ;)

File: app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                modifier = Modifier.constrainAs(valueRef) {
+                    width = Dimension.fillToConstraints
+                    start.linkTo(parent.start)
+                    end.linkTo(imageRef.start)
+                    top.linkTo(labelRef.bottom)
+                },
+            )
+            Image(
+                painter = painterResource(id = R.drawable.baseline_upload_file_24),
+                contentDescription = "Select Certifcate",
```
Comment (Chrylo) [inline app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt:90]: Typo "Certificate"

File: app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                },
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+fun FileSelectorSettingPreview() {
+    FileSelectorSettingView(label = "Certificate", value = "CA.pem") {
```
Comment (Chrylo) [inline app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt:105]: Typo

**Review by Chrylo — CHANGES_REQUESTED** [review]

File: app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt (hunk 7 above)
Comment (wba2hi) [reply app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt:105]: Uhm,... I don't see the typo, please elaborate here

File: app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt (hunk 4 above)
Comment (wba2hi) [reply app/src/test/kotlin/org/eclipse/kuksa/testapp/model/ConnectionInfoTest.kt:30]: Added "test" module to share test specific stuff like the Kotest NamedTags or Kotest configuration.
Also added the run config for app:UnitTests

File: app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt (hunk 7 above)
Comment (Chrylo) [reply app/src/main/kotlin/org/eclipse/kuksa/testapp/databroker/view/FileSelectorSettingView.kt:105]: Magic :O

**Review by Chrylo — APPROVED** [review]

**Comment by Chrylo** [comment]
LGTM
- Ran Unit Tests 
- Testes cert upload + connection

**Comment by wba2hi** [comment]
> So it works but if I use a file that does not work it will just timeout. As user I would want to know that it times out because of my wrong file or because I didn't specify one. Otherwise it looks good.

Unfortunately we don't get that info. We only get the "ConnectionState" (IDLE, CONNECTED, CLOSED,...) to verify if the channel to the DataBroker is connected. We can roll that part up again and see if we can improve the "how to connect"-logic but this would be part of a separate ticket.

**Comment by lukasmittag** [comment]
Understood okay :)

**Review by lukasmittag — APPROVED** [review]
LGTM. What I did:
* tested uploading certificates and using them (false and right ones)

**Pull request merged**